core/src/main/java/gyro/core/reference/FinderReferenceResolver.java

```diff
@@ -19,23 +19,29 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 import java.util.stream.Collectors;
 
 import gyro.core.GyroException;
 import gyro.core.Type;
+import gyro.core.auth.CredentialsSettings;
 import gyro.core.finder.Finder;
 import gyro.core.finder.FinderField;
 import gyro.core.finder.FinderSettings;
 import gyro.core.finder.FinderType;
 import gyro.core.resource.DiffableInternals;
 import gyro.core.resource.Resource;
 import gyro.core.scope.Scope;
+import gyro.lang.ast.value.ReferenceNode;
 
 @Type("external-query")
 public class FinderReferenceResolver extends ReferenceResolver {
 
     @Override
-    public Object resolve(Scope scope, List<Object> arguments) {
+    public Object resolve(ReferenceNode node, Scope scope, List<Object> arguments) {
```

Comment:
If we go the route of using `node` to parse options, similar to how Directives work, then we probably should think about calculating `arguments` from `node` and not passing it in. These two parameters are at odds with each other.

Reply:
Refactored code to use a common class and a common processor to handle arguments and options